test/e2e/kubectl/kubectl.go

```diff
@@ -1079,6 +1098,109 @@ metadata:
 			err = createApplyCustomResource(validArbitraryCR, f.Namespace.Name, "test-cr", crd)
 			framework.ExpectNoError(err, "creating custom resource")
 		})
+
+		ginkgo.It("should detect unknown metadata fields in both the root and embedded object of a CR", func() {
+			ginkgo.By("prepare CRD with x-kubernetes-embedded-resource: true")
+			opt := func(crd *apiextensionsv1.CustomResourceDefinition) {
+				props := &apiextensionsv1.JSONSchemaProps{}
+				if err := yaml.Unmarshal(schemaFooEmbedded, props); err != nil {
+					framework.Failf("failed to unmarshal schema: %v", err)
+				}
+				crd.Spec.Versions = []apiextensionsv1.CustomResourceDefinitionVersion{
+					{
+						Name:    "v1",
+						Served:  true,
+						Storage: true,
+						Schema: &apiextensionsv1.CustomResourceValidation{
+							OpenAPIV3Schema: props,
+						},
+					},
+				}
+			}
+
+			group := fmt.Sprintf("%s.example.com", f.BaseName)
+			testCRD, err := crd.CreateMultiVersionTestCRD(f, group, opt)
+			if err != nil {
+				framework.Failf("failed to create test CRD: %v", err)
+			}
+			defer testCRD.CleanUp()
+
+			ginkgo.By("sleep for 10s to wait for potential crd openapi publishing alpha feature")
+			time.Sleep(10 * time.Second)
+
+			ginkgo.By("attempting to create a CR with unknown metadata fields at the root level")
+			schema := schemaForGVK(schema.GroupVersionKind{Group: testCRD.Crd.Spec.Group, Version: testCRD.Crd.Spec.Versions[0].Name, Kind: testCRD.Crd.Spec.Names.Kind})
+			framework.ExpectNotEqual(schema, nil, "retrieving a schema for the crd")
+			embeddedCRPattern := `
+
+{%s,
+  "spec": {
+          "template": {
+      	          "apiVersion": "foo/v1",
+      	 	  "kind": "Sub",
+      		  "metadata": {
+			%s
+      		        "name": "subobject"
+      		  },
+      		  "spec": null
+          }
+  }
+}`
```

Comment:
nit: indentation/whitespace on this is mixing tabs and spaces and is pretty hard to read... indent with 2-spaces consistently?